packages/caravan-clients/src/client.test.ts

```diff
@@ -1823,4 +1825,327 @@ describe("BlockchainClient", () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.MEMPOOL,
+        network: Network.MAINNET,
+      });
+
+      await expect(client.getAddressTransactionHistory(testAddress, 0))
+        .rejects.toThrow('Count must be between 1 and 100000');
+      
+      await expect(client.getAddressTransactionHistory(testAddress, 100001))
+        .rejects.toThrow('Count must be between 1 and 100000');
+    });
+
+    it('should validate skip parameter', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.MEMPOOL,
+        network: Network.MAINNET,
+      });
+
+      await expect(client.getAddressTransactionHistory(testAddress, 10, -1))
+        .rejects.toThrow('Skip must be non-negative');
+    });
+  });
+
+  describe('PRIVATE client', () => {
+    let client: BlockchainClient;
+
+    beforeEach(() => {
+      client = new BlockchainClient({
+        type: ClientType.PRIVATE,
+        network: Network.MAINNET,
+        client: {
+          url: 'http://localhost:8332',
+          username: 'user',
+          password: 'pass',
+          walletName: 'test-wallet',
+        },
+      });
+    });
+
+    it('should throw error for private clients', async () => {
+      await expect(client.getAddressTransactionHistory(testAddress))
+        .rejects.toThrow('Use getWalletTransactionHistory for private clients');
+    });
+
+    it('should throw error for private clients with array of addresses', async () => {
+      await expect(client.getAddressTransactionHistory([testAddress, testAddress2]))
+        .rejects.toThrow('Use getWalletTransactionHistory for private clients');
+    });
+  });
+
+  describe('PUBLIC client - single address', () => {
+    it('should use correct endpoint for MEMPOOL', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.MEMPOOL,
+        network: Network.MAINNET,
+      });
+
+      const mockResponse = [
+        { 
+          txid: 'tx1', 
+          fee: 1000,
+          time: 1234567890,
+          version: 1,
+          locktime: 0,
+          vin: [],
+          vout: [],
+          size: 250,
+          weight: 1000,
+          status: { confirmed: true }
+        }
+      ];
+      const mockGet = vi.fn().mockResolvedValue(mockResponse);
+      client.Get = mockGet;
+
+      const result = await client.getAddressTransactionHistory(testAddress, 20, 40);
+
+      expect(mockGet).toHaveBeenCalledWith(`/address/${testAddress}/txs?count=20&skip=40`);
+      expect(result).toHaveLength(1);
+    });
+
+    it('should use correct endpoint for BLOCKSTREAM', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.BLOCKSTREAM,
+        network: Network.MAINNET,
+      });
+
+      const mockResponse = [
```

Comment:
no need to re-create the same fixture. Would the tests fail if you didn't mock the results? Unless you're testing any mutations that the function does, it shouldn't be necessary to return a full object.